{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "4bfb383f-59c6-4629-9b99-40a08d0ecb4a",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pathlib import Path\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "9eabfdf1-0f7f-40fa-9e42-188b5c75ae4a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>population</th>\n",
       "      <th>meanEstTotalCount</th>\n",
       "      <th>EstRNA_yield_ng</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NeuN+</td>\n",
       "      <td>4.807840e+06</td>\n",
       "      <td>5745.368999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PU1+</td>\n",
       "      <td>2.289774e+05</td>\n",
       "      <td>273.627996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SOX10+</td>\n",
       "      <td>9.454005e+05</td>\n",
       "      <td>1129.753581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SOX2+</td>\n",
       "      <td>2.616858e+05</td>\n",
       "      <td>312.714571</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  population  meanEstTotalCount  EstRNA_yield_ng\n",
       "0      NeuN+       4.807840e+06      5745.368999\n",
       "1       PU1+       2.289774e+05       273.627996\n",
       "2     SOX10+       9.454005e+05      1129.753581\n",
       "3      SOX2+       2.616858e+05       312.714571"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Quick aside: estimating RNA yields.\n",
    "\n",
    "# load in the tidied, long-format FANS data\n",
    "df_path = Path(\"tidied_FANS_data/exp383_tidy_FANS_data_long.csv\")\n",
    "df = pd.read_csv(df_path)\n",
    "df.head()\n",
    "\n",
    "populationStats = (\n",
    "    df.groupby([\"population\", \"count_statistic\"])[\"count_value\"]\n",
    "        .agg([\"count\", \"mean\", \"std\", \"min\", \"max\"])\n",
    "        .sort_values(by=[\"count_statistic\", \"population\"])\n",
    "        .reset_index()\n",
    ")\n",
    "\n",
    "# I found a table in a review articles with estimated picogram quantities of RNA\n",
    "# in eukaryotic cells.\n",
    "# I also found a comparison of single-cell and single nucleus transcriptomes and found\n",
    "# that 20% of RNA is nuclear. We can work out some estimated yields using this info.\n",
    "\n",
    "nucRNAprop = 0.2/2\n",
    "estRNAperCell_ng = 23.9e-3/2\n",
    "estRNAperNuc_ng = estRNAperCell_ng  * nucRNAprop\n",
    "\n",
    "# make new df values\n",
    "pops = list(populationStats[\"population\"].unique())\n",
    "means= list(populationStats['mean'][0:4])\n",
    "ngs  = [i*estRNAperNuc_ng for i in means]\n",
    "    \n",
    "estRNAYields = pd.DataFrame(\n",
    "    {\"population\" : pops,\n",
    "     \"meanEstTotalCount\": means,\n",
    "     \"EstRNA_yield_ng\": ngs    \n",
    "    }\n",
    ")\n",
    "\n",
    "estRNAYields.style.format({\n",
    "    \"meanEstTotalCount\": \"{:,.0f}\",  \n",
    "    \"EstRNA_yield_ng\"  : \"{:,.1f}\"\n",
    "})\n",
    "\n",
    "estRNAYields\n",
    "\n",
    "# analysis doesnt make sense as we have a proportion of nuclear rna \n",
    "# by proportion of all transcripts rather than by mass"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:generalPython] *",
   "language": "python",
   "name": "conda-env-generalPython-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
